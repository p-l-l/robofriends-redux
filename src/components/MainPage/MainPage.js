import React, { Component } from 'react';

import CardList from '../CardList/CardList';
import SearchBox from '../SearchBox';
import Scroll from '../Scroll';
import ErrorBoundry from '../ErrorBoundry';

import './MainPage.css';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
      return this.props.robots.filter(robot => {
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
      })
  }

  render() {
    const { robots, onSearchChange, isPending } = this.props;

    return (
      <div className='tc'>
        <h1 className='f1'>RoboFriends</h1>
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundry>
              <CardList robots={this.filteredRobots(robots)} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
