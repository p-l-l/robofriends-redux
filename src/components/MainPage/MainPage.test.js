import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';



describe('MainPage component', () => {

    let wrapper;

    // Before each test, we run this to load the Component...
    // Note : This is not efficient here, but we use beforeEach anyway to show what it can do.
    beforeEach(() => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [],
            searchField: '',
            isPending: false
        }
        wrapper = shallow(<MainPage {...mockProps}/>)
    })

    // After each test, we unmount the Component to make sure tests remains independant from one another.
    afterEach(() => {
        wrapper.unmount();
    });

    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('filters robots correctly', () => {
        const mockProps2 = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com'
            }],
            searchField: 'john',
            isPending: false
        }
        
        const wrapper2 = shallow(<MainPage {...mockProps2}/>);
    
        expect(wrapper2.instance().filteredRobots()).toEqual([{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }]);
    })

    it('filters robots correctly 2', () => {
        const mockProps3 = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com'
            }],
            searchField: 'a',
            isPending: false
        }
        const filteredRobots = []
        const wrapper3 = shallow(<MainPage {...mockProps3}/>);
    
        expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots);
    })

});