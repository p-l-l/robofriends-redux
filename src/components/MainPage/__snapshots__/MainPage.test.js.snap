// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component Renders correctly 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
